### PR TITLE
[Storage]Update Storage API version to 2016-12-01 with file encryption service

### PR DESCRIPTION
**Pull request by JasonYang-MSFT**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [x] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.

### CODE REVIEW

File: arm-storage/2016-12-01/swagger/storage.json [hunk 1]
```diff
@@ -0,0 +1,1318 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "StorageManagement",
+    "description": "The Azure Storage Management API.",
+    "version": "2016-12-01",
+    "x-ms-code-generation-settings": {
```
Comment (fearthecowboy) [inline arm-storage/2016-12-01/swagger/storage.json]: Just tagging what changed between 2016-05-01 and this one. 

Added:
``` json 
 "version": "2016-12-01",
    "x-ms-code-generation-settings": {
      "name": "StorageManagementClient"
    }
```

File: arm-storage/2016-12-01/swagger/storage.json [hunk 2]
```diff
@@ -0,0 +1,1318 @@
...
+      },
+      "description": "A service that allows server-side encryption to be used."
+    },
+    "EncryptionServices": {
+      "properties": {
+        "blob": {
+          "$ref": "#/definitions/EncryptionService",
+          "description": "The encryption function of the blob storage service."
+        },
+        "file": {
```
Comment (fearthecowboy) [inline arm-storage/2016-12-01/swagger/storage.json]: Tagging what changed 

Added
``` json
 },
        "file": {
          "$ref": "#/definitions/EncryptionService",
          "description": "The encryption function of the file storage service."
        },
        "table": {
          "$ref": "#/definitions/EncryptionService",
          "readOnly": true,
          "description": "The encryption function of the table storage service."
        },
        "queue": {
          "$ref": "#/definitions/EncryptionService",
          "readOnly": true,
          "description": "The encryption function of the queue storage service."
        }
```

Comment (fearthecowboy) [inline arm-storage/2016-12-01/swagger/storage.json]: Confirming: the settings for `table` and `queue` are intended to be `readOnly`?

How do they get their state changed?

**Comment by fearthecowboy** [comment]
@JasonYang-MSFT Am I correct in that the change was just those two simple things?

File: arm-storage/2016-12-01/swagger/storage.json (hunk 2 above)
Comment (JasonYang-MSFT) [inline arm-storage/2016-12-01/swagger/storage.json]: Yes, it is intended to be readOnly. 
Encryption state for table and queue are controlled per DC, not settable per account.

**Comment by JasonYang-MSFT** [comment]
@fearthecowboy 
Yes, this is it for storage service encryption feature.

**Review by fearthecowboy — APPROVED** [review]
Looks good to me then. 🤠

**Pull request merged**